File: solutions/java/resistor-color/2/src/main/java/ResistorColor.java
import java.util.Arrays;
import java.util.List;

class ResistorColor {
    static final private List<String> COLORS = List.of(
        new String[]{
            "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"
        }
    );
    
    static int colorCode(String color) {
        return COLORS.indexOf(color);
    }

    String[] colors() {
        return COLORS.toArray(new String[0]);
    }
}

File: solutions/typescript/resistor-color/2/resistor-color.ts
export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white'
] as const;

export type Color = typeof COLORS[number];

export const colorCode = (color:Color) => COLORS.indexOf(color);